components/brave_extension/extension/brave_extension/background/today/feedToData.ts

```diff
@@ -0,0 +1,260 @@
+// Copyright (c) 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+import { formatDistanceToNow } from 'date-fns'
+import hashStrings from '../../../../../common/hashStrings'
+
+function shuffleArray(array: Array<any>) {
```

Comment:
Is there any spec for the shuffle? ex: what type of shuffle algo is this (is that good enough?)

Reply:
nope, just random AFAIK. It's only 4 articles per page that are selected at random from the past 48 hours. That happens here:

```
const randomArticles = take(articles, 4, isArticleWithin48Hours, true)
```